### PR TITLE
fix User has to click twice to select currency

### PR DESCRIPTION
**Pull request by tienifr**
### Fixed Issues

$ https://github.com/Expensify/App/issues/41922
PROPOSAL: https://github.com/Expensify/App/issues/41922#issuecomment-2112071705

### Tests

1. Open app.
2. Click on FAB > Submit expense > Click on the currency to open the currency selector.
3. Long press on a currency located around the bottom of the list.
4. The currency gets selected on the first click
- [x] Verify that no errors appear in the JS console

### QA Steps

1. Open app.
2. Click on FAB > Submit expense > Click on the currency to open the currency selector.
3. Long press on a currency located around the bottom of the list.
4. The currency gets selected on the first click
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/113963320/54f5888c-ec23-4869-9b85-8eac30ef258d

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/113963320/4153d5e9-5f5e-440d-bf7f-d2779f65ec4f

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/113963320/234eed3e-832d-4d75-8a16-992e13a96344

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/113963320/a1556967-bc48-49df-991f-86ac32413373

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/113963320/c4051b90-d004-4820-8c3b-6aabd7a91f1d

</details>

### CODE REVIEW

**Comment by jayeshmangwani** [comment]
@tienifr Update the Test step And mention that we need to long press the currency, and also Android and Android chrome Videos are missing in the checklist

**Comment by jayeshmangwani** [comment]
@tienifr I am able to reproduce the original issue in Android chrome, would you confirm if its working for you.

**Comment by jayeshmangwani** [comment]
@tienifr bump for the update

**Comment by jayeshmangwani** [comment]
Thanks for the update

**Comment by jayeshmangwani** [comment]
@tienifr Are you able to find the cause of why it's happening for Android Chrome?

**Comment by jayeshmangwani** [comment]
> so when the user long presses on any option, it will trigger onLongPress and do like we do with onPress. It can just apply this in Android Chrome. What do you think?

What's the reason why it's working for the mWeb safari and not the mWeb Chrome? It should be worked the same way in both Safari and Chrome

**Comment by jayeshmangwani** [comment]
Thanks, I will also try to look into this today

**Comment by jayeshmangwani** [comment]
@tienifr No, But I am not confident in this [solution either](https://github.com/Expensify/App/pull/42380#issuecomment-2139234610), I would suggest we should close this PR and open the original issue for the new proposals that also solve with the Android chrome too 
cc: @Julesssss

**Comment by Julesssss** [comment]
> Therefore, I propose that we address this issue across all platforms as it is a straightforward fix. Subsequently, we can create a new issue specifically to resolve the long-press problem in Android Chrome.

I mostly agree with this solution. Android mWeb seems like an outstanding issue that is related. Perhaps we could hold this temporarily to see if the solution to Android mWeb ends up giving us a similar solution here?

**Comment by jayeshmangwani** [comment]
> Perhaps we could hold this temporarily to see if the solution to Android mWeb ends up giving us a similar solution here?

Are you suggesting that we should proceed with this PR and create a new issue for the Android Chrome's long-press problem?

**Comment by Julesssss** [comment]
> Are you suggesting that we should proceed with this PR and create a new issue for the Android Chrome's long-press problem?

Yeah I see two options: 
- Finish this PR, isolated to mWeb
- Hold this PR, wait for a proposal for the native permission (get approval for the library, etc)

I prefer the first, because it could take a while to prioritise and get a library solution implemented. Does that sound good?

**Comment by jayeshmangwani** [comment]
Yes, that sounds good. We will finish this PR and for Android chrome will open the other discussion

**Comment by jayeshmangwani** [comment]
Yes @tienifr , Will review in next 2 hours, I am on other PR rn and then review this

**Comment by jayeshmangwani** [comment]
https://github.com/Expensify/App/blob/59a37c864a2cb41f2b5f9642255197c19e9245e3/src/components/SelectionList/InviteMemberListItem.tsx#L59


https://github.com/Expensify/App/blob/59a37c864a2cb41f2b5f9642255197c19e9245e3/src/components/SelectionList/RadioListItem.tsx#L37

https://github.com/Expensify/App/blob/59a37c864a2cb41f2b5f9642255197c19e9245e3/src/components/SelectionList/TableListItem.tsx#L56

https://github.com/Expensify/App/blob/59a37c864a2cb41f2b5f9642255197c19e9245e3/src/components/SelectionList/UserListItem.tsx#L62

https://github.com/Expensify/App/blob/59a37c864a2cb41f2b5f9642255197c19e9245e3/src/components/SelectionList/Search/ReportListItem.tsx#L137

https://github.com/Expensify/App/blob/59a37c864a2cb41f2b5f9642255197c19e9245e3/src/components/SelectionList/Search/TransactionListItem.tsx#L45

@tienifr we don't need to pass the `shouldPreventDefaultFocusOnSelectRow` prop to above components, we can get rid of unused prop here, and please test once all the places where we have used the `BaseListItem` to confirm that these change doesn't cause regression to the other components where we are using the `BaseListItem`

**Comment by jayeshmangwani** [comment]
And also remove the unused props

**Comment by jayeshmangwani** [comment]
@tienifr Maybe its for my simulator only, but long pressing does nothing for me on mWeb safari. can you please give a try on your side does long press select the currency for you on mWeb safari ?

**Comment by jayeshmangwani** [comment]
Yes sure, here's the video

https://github.com/Expensify/App/assets/35371050/ec7c4226-8d4f-4cc7-90f8-3aa64075f9b5

**Comment by jayeshmangwani** [comment]
Yes tried, long press doesn't do anything for me

**Comment by jayeshmangwani** [comment]
> Here is from my side:

I have tried with the few simulators, long press doesn't work for me on mWeb safari.

From your video I can say you long-pressed currency for around 2 seconds, I would say try with few more seconds, e.g. long-press for 5 seconds,For me pressing few more seconds is not working on the mWeb safari but it is working for web

**Comment by jayeshmangwani** [comment]
> I tried with other components in mWeb safari, such as "Back" button. When you long press for 2 seconds, it triggers onPress event

Are you able to reproduce the issue that I have mentioned [here](https://github.com/Expensify/App/pull/42380#issuecomment-2155159220) ?

**Comment by jayeshmangwani** [comment]
> If I long press more than 5 seconds, it does nothing

Thanks for confirming

@Julesssss when we press currency option for longer than 5 seconds, no event is triggered in Safari on mWeb, tienifr confirmed that [this behavior](https://github.com/Expensify/App/pull/42380#issuecomment-2157104878) is consistent across the app, Should we proceed with the PR, ignoring case where the user presses for more than 5-6 seconds?

**Comment by Julesssss** [comment]
> Should we proceed with the PR, ignoring case where the user presses for more than 5-6 seconds?

Ah damn, sorry I should have noticed earlier 😞 

I think not, if the behaviour is consistent and with this being an odd use case I think we should not move forward. If anyone disagrees please let me know.

**Comment by Julesssss** [comment]
I understood this sentence to mean that long pressing components on iOS safari to be consistent throughout the app: 

> tienifr confirmed that https://github.com/Expensify/App/pull/42380#issuecomment-2157104878 is consistent across the app

Is that not the case?

**Comment by Julesssss** [comment]
Thanks for explaining 👍 

> Should we proceed with the PR, ignoring case where the user presses for more than 5-6 seconds?

So yes, I think that is totally fine.

**Comment by jayeshmangwani** [comment]
> So yes, I think that is totally fine.

@Julesssss Should we move forward with this PR ?

**Comment by Julesssss** [comment]
@jayeshmangwani yep

File: src/pages/workspace/taxes/WorkspaceTaxesPage.tsx
Comment (jayeshmangwani) [inline src/pages/workspace/taxes/WorkspaceTaxesPage.tsx:303]: @tienifr IMO we should not remove the `shouldPreventDefaultFocusOnSelectRow`  prop from the SelectionList Component this will cause the unexpected result as this was added for some other reason

https://github.com/Expensify/App/blob/0c3b262e0cc21aee142e489f3c8aaa8d26189e1e/src/components/SelectionList/BaseSelectionList.tsx#L307-L309

https://github.com/Expensify/App/blob/0c3b262e0cc21aee142e489f3c8aaa8d26189e1e/src/components/SelectionList/BaseSelectionList.tsx#L315-L317

https://github.com/Expensify/App/blob/0c3b262e0cc21aee142e489f3c8aaa8d26189e1e/src/components/SelectionList/BaseSelectionList.tsx#L407

**Comment by jayeshmangwani** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/35371050/eb2fdf1d-ebe0-4f61-92ea-f0bed508b22d




</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/35371050/4d871cfb-3da2-4d6a-bca8-3268749d307f




</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/35371050/c216a1c2-231e-4ea4-bf78-d1bec0148ffb




</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/35371050/bd187be3-7523-4e92-9535-2d4fefc1a6b6




</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/35371050/cd149240-a738-4d5c-96ee-4c18cfb722b5




</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/35371050/1c85d441-252a-4540-99c5-7bdf175e2fd5




</details>

**Review by jayeshmangwani — APPROVED** [review]
LGTM 🚀

**Review by Julesssss — APPROVED** [review]

**Pull request merged**
